roundtripper/github.go

```diff
@@ -55,14 +56,15 @@ func makeTokenAccessor(accessTokens []string) tokenAccessor {
 type roundRobinAccessor struct {
 	accessTokens []string
 	counter      int64
+	mu           sync.Mutex
 }
 
-func (roundRobin *roundRobinAccessor) next() (*oauth2.Token, error) {
+func (roundRobin *roundRobinAccessor) next() string {
+	roundRobin.mu.Lock()
+	defer roundRobin.mu.Unlock()
```

Comment:
Why do we need a lock?

Reply:
Without lock, the fn is not doing what we expect. E.g: thread1 increments `counter`, then context switches and thread2 also increments `counter`. Then both the threads access the same index through `index := c % int64(l)`.

Although not having a lock won't cause any issues, its not technically round robin.